### PR TITLE
Make CRD gen check ext versions for in scope. Update CSRs with correc…

### PR DESCRIPTION
**Pull request by dhower-qc**
…t version info

### CODE REVIEW

**Comment by dhower-qc** [comment]
fixes #163

File: arch/csr/mseccfg.yaml
Comment (james-ball-qualcomm) [inline arch/csr/mseccfg.yaml:1]: I see from this change that a space is required between the operator (>= in this case) and the version number. This seems like a common bug. Can you remove this requirement to have a space or have the schema check for it? Or have an error generated when creating backends or parsing the database?

File: arch/csr/medelegh.yaml
Comment (james-ball-qualcomm) [inline arch/csr/medelegh.yaml:1]: Don't mdelegh and mseccfgh need the same "definedBy" as you added for mstatush? Namely,

definedBy:
    name: Sm
    version: ">= 1.12"

File: backends/crd_doc/templates/crd.adoc.erb
Comment (james-ball-qualcomm) [inline backends/crd_doc/templates/crd.adoc.erb:1]: Can you explain why you had to switch from in_scope_extensions() to in_scope_ext_reqs()? I want to add comments to the code to explain why one is used vs. the other for future reference.

File: lib/arch_obj_models/extension.rb
Comment (james-ball-qualcomm) [inline lib/arch_obj_models/extension.rb:1]: You have a typo: implemente**s**d_instructions. Did you test this?

**Review by james-ball-qualcomm — CHANGES_REQUESTED** [review]
Mostly just questions to better understand your changes but one typo found that is a real bug.

File: backends/crd_doc/templates/crd.adoc.erb
Comment (dhower-qc) [reply backends/crd_doc/templates/crd.adoc.erb:1]: When you just ask for the extension list, there is no version info -- it's just a list of extension info. You need the ExtensionRequirement objects (which combines and Extension with a Requirement) in order to filter by, e.g., a "= 1.11" requirement.

File: lib/arch_obj_models/extension.rb
Comment (dhower-qc) [reply lib/arch_obj_models/extension.rb:1]: I did test it (and you can see it's passing the regressions). I added this function thinking I was going to use it, but then changed my mind. It's still a useful function, which is why I left it. Good job finding the typo.

File: arch/csr/medelegh.yaml
Comment (dhower-qc) [reply arch/csr/medelegh.yaml:1]: Fixed medelegh (>= 1.13). mseccfgh already says >= 1.12

**Pull request merged**